src/renderer/tvgWgCanvas.cpp

```diff
@@ -52,25 +52,33 @@ WgCanvas::~WgCanvas()
 {
 #ifdef THORVG_WG_RASTER_SUPPORT
     auto renderer = static_cast<WgRenderer*>(Canvas::pImpl->renderer);
-    renderer->target(nullptr, 0, 0);
+    renderer->target(nullptr, (WGPUSurface)nullptr, 0, 0, nullptr);
 #endif
 }
 
 
-Result WgCanvas::target(void* instance, void* surface, uint32_t w, uint32_t h, void* device) noexcept
+Result WgCanvas::target(void* instance, void* target, uint32_t w, uint32_t h, int type, void* device) noexcept
 {
 #ifdef THORVG_WG_RASTER_SUPPORT
     if (Canvas::pImpl->status != Status::Damaged && Canvas::pImpl->status != Status::Synced) {
         return Result::InsufficientCondition;
     }
 
-    if (!instance || !surface || (w == 0) || (h == 0)) return Result::InvalidArguments;
+    if (!instance || !target || (w == 0) || (h == 0)) return Result::InvalidArguments;
 
     //We know renderer type, avoid dynamic_cast for performance.
     auto renderer = static_cast<WgRenderer*>(Canvas::pImpl->renderer);
     if (!renderer) return Result::MemoryCorruption;
 
-    if (!renderer->target((WGPUInstance)instance, (WGPUSurface)surface, w, h, (WGPUDevice)device)) return Result::Unknown;
+    // surface
+    if (type == 0) {
+        if (!renderer->target((WGPUInstance)instance, (WGPUSurface)target, w, h, (WGPUDevice)device)) return Result::Unknown;
```

Comment:
You can cast the target in the renderer engine, not in the canvas. There's no need for an if-else here.